prody2/viewers/viewer_gnm.py

```diff
@@ -0,0 +1,266 @@
+from pwem.emlib import MDL_NMA_ATOMSHIFT
+from pwem.objects import SetOfNormalModes
+
+from continuousflex.protocols import FlexProtNMA
+from continuousflex.viewers.nma_plotter import FlexNmaPlotter
+
+from prody2.protocols import ProDyGNM
+
+import os
+
+import prody
+from prody.utilities.drawtools import IndexFormatter
+from matplotlib.pyplot import *
+
+OBJCMD_NMA_PLOTDIST = "Plot distance profile"
+OBJCMD_NMA_VMD = "Display VMD animation"
+
+
+class ProDyGNMViewer(ProtocolViewer):
+    """ Visualization of results from the GNM-NMA protocol.    
+        Normally, modes with high collectivity and low 
+        score are preferred.
+    """
+    _label = 'viewer gnm'
+    _targets = [ProDyGNM]
+    _environments = [DESKTOP_TKINTER, WEB_DJANGO]
+
+#     def setProtocol(self, protocol):
+#         ProtocolViewer.setProtocol(self, protocol)
+#         inputPdb = protocol.inputStructure.get()
+#         self.isEm.set(inputPdb.getPseudoAtoms())
+        
+    def _defineParams(self, form):
+
+        if isinstance(self.protocol, SetOfNormalModes):
+            protocol_path = os.path.dirname(os.path.dirname(self.protocol[1].getModeFile()))
+            vmdFiles = protocol_path + "/extra/animations/"
+            nmdFile = protocol_path + "/modes.nmd"
+        else:
+            vmdFiles = self.protocol._getExtraPath("animations")
+            nmdFile = self.protocol._getPath("modes.nmd")
+
+        form.addSection(label='Visualization')
+  
+        form.addParam('displayModes', LabelParam,
+                      label="Display output Normal Modes?", important=True)
+        form.addParam('displayMaxDistanceProfile', LabelParam,
+                      label="Plot max distance profile?",
+                      help="Maximum unitary shift of each atom or pseudoatom over all computed modes.") 
+
+        form.addParam('displaySqFlucts', LabelParam,
+                      label="Plot MSF from all the computed modes?",
+                      help="The MSF residues are found on the diagonal of "
+                           "the covariance matrix and describe the residue mobility or flexibility.")
+
+        group = form.addGroup('Single mode')  
+        group.addParam('modeNumber', IntParam, default=7,
+                      label='Mode number')
+        group.addParam('displayVmd', LabelParam,
+                      condition=os.path.isdir(vmdFiles),
+                      label='Display mode animation with VMD?') 
+        group.addParam('displayDistanceProfile', LabelParam, default=False,
+                      label="Plot mode distance profile?",
+                      help="Unitary shift of each atom or pseudoatom along the mode that is requested to be animated.")
+        group.addParam('displaySingleSqFluct', LabelParam, default=False,
+                      label="Plot mode square fluctuation?",
+                      help="Shows the single mode Square Fluctuation")
+        group.addParam('cumulSqFluct', BooleanParam, default=False,
```

Comment:
How is this different? The help probably needs to be better. I understand you are saying cumulative up to the number selected by looking at the code but from just the help I wasn't sure. 

Anyhow, I'm not sure if this is a useful option. It could be better for the user to put in a range of modes if they want the cumulative. The calculation is actually the same one except for how you are doing the indexing.